java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -50,6 +50,11 @@
     this(type, rows, nullCount, hostDataBuffer, hostValidityBuffer, null);
   }
 
+  HostColumnVector(ai.rapids.cudf.DataType type, long rows, Optional<Long> nullCount,
```

Comment:
In HostColumnVector there is an abstract class `DataType`. So I am using fully qualified name in this file. Please let me know if you prefer any other name to represent DataType class(with DType + scale) if this is confusing. I wasn't sure what else would represent `DataType` : ) 
https://github.com/rapidsai/cudf/blob/branch-0.17/java/src/main/java/ai/rapids/cudf/HostColumnVector.java#L1545

Reply:
You don't need the fully qualified name imo. That's what Jason is saying I think

Reply:
Ah, sorry I totally missed that `HostColumnVector` already has a `DataType`.  It would be ideal to not collide with this name, but I'm not coming up with a better alternative name offhand.

Reply:
Thanks! Will keep it as it is for now and update it later if/when we come up with better name.

Reply:
I'm sorry I haven't really looked into this change, but why can't we change and tweak the one that exists?

Reply:
I agree with @kuhushukla can we combine the two? 

```
class DataType {
   DType typeId
   int scale 
   def isNullable()
   def getSize() // may be return size in bytes for primitive types
}

class NestedType extends DataType {
   def getNumChildren()
   def getChild(int index) 
}

class StructType extends NestedType
class ListType extends NestedType

```

Reply:
I agree that we need to figure out long term how we want to handle nested schemas.  Currently we have a Schema class that is used with CSV parsing, DType and DataType.  I think this needs to be something where someone gathers all of the requirements and comes up with a design for what we really want this to look like.  I don't think that has been done yet.

Reply:
I also want to add that part of the reason we did it this way was to avoid most issues with source incompatibility. If we do what is being proposed haphazardly we will break anyone using the APIs so we need to be careful.  The current changes should be very close to source compatible.  Minor changes if any.  We can get away with some of that because we have singleton objects.

The other issue I want to call out is that cudf itself has no schema equivalent right now.  So if we do combine them all together then we will have no clear mapping between this java object and a corresponding C++ object.  I am not sure that is ideal.